dgraph-modus-example/backend/main.go

```diff
@@ -0,0 +1,201 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/hypermodeinc/modus/sdk/go/pkg/http"          // Modus HTTP library for making API requests
+	"github.com/hypermodeinc/modus/sdk/go/pkg/models"        // Modus models library for AI models
+	"github.com/hypermodeinc/modus/sdk/go/pkg/models/openai" // Modus OpenAI model integration
+)
+
+// MovieDetails represents the structure of a movie's details retrieved from Dgraph
+type MovieDetails struct {
+	Name string `json:"name@en"` // JSON tag maps Dgraph's "name@en" field
+}
+
+// FetchMoviesWithPaginationAndSearch fetches a paginated list of movies from Dgraph with optional search functionality
+// This function showcases the power of Dgraph in efficiently handling graph-based queries and filtering
+// through interconnected datasets, such as movies, genres, actors, and directors.
+func FetchMoviesWithPaginationAndSearch(page int, search string) (string, error) {
+	offset := (page - 1) * 10 // Calculate offset for pagination (10 movies per page)
+
+	// The Dgraph query retrieves movies with pagination, sorting by release date in descending order.
+	// Dgraph's graph structure allows linking genres, actors, and directors, enabling a rich and connected query.
+	query := fmt.Sprintf(`{
+		movies(func: has(initial_release_date), first: 10, offset: %d, orderdesc: initial_release_date) %s {
+			uid
+			name@en
+			initial_release_date
+			genre {
+				name@en
+			}
+			starring {
+				performance.actor {
+					name@en
+				}
+			}
+			directed_by: director.film {
+				name@en
+			}
+		}
+	}`, offset, buildSearchFilter(search))
+
+	return executeDgraphQuery(query)
+}
+
+// buildSearchFilter generates a search filter to include in the Dgraph query
+func buildSearchFilter(search string) string {
+	if search == "" {
+		return ""
+	}
+
+	// The @filter clause matches the search term against movie titles, genres, actor names, and director names.
+	// This demonstrates Dgraph's flexibility in querying multiple fields simultaneously.
+	return fmt.Sprintf(`@filter(
+		anyoftext(name@en, "%[1]s") OR
+		anyoftext(genre.name@en, "%[1]s") OR
+		anyoftext(starring.performance.actor.name@en, "%[1]s") OR
+		anyoftext(directed_by.name@en, "%[1]s")
+	)`, search)
+}
```

Comment:
Dgraph does not support this to my knowledge.
Have you tested a request in play.dgraph.io using this notation with dot: "genre.name@en" or 
starring.performance.actor.name@en
Have you seen that in Dgraph doc?

Reply:
Ahh thank you, didn't realize because there were no errors. I just pushed a fix, let me know if there are still any issues (the new query works in ratel)